syslog_ng/config/file.sls

```diff
@@ -0,0 +1,30 @@
+# -*- coding: utf-8 -*-
+# vim: ft=sls
+
+{#- Get the `tplroot` from `tpldir` #}
+{%- set tplroot = tpldir.split('/')[0] %}
+{%- set sls_package_install = tplroot ~ '.package.install' %}
+{%- set sls_reload_service  = tplroot ~ '.service.reload' %}
+{%- from tplroot ~ "/map.jinja" import syslog_ng with context %}
+{%- from tplroot ~ "/libtofs.jinja" import files_switch with context %}
+
+include:
+  - {{ sls_package_install }}
+  - {{ sls_reload_service }}
+
+syslog_ng-config-file-file-managed:
+  file.managed:
+    - name:     {{ syslog_ng.config_file }}
+    - user:     root
+    - group:    root
+    - template: jinja
+    - source: {{ files_switch(['syslog-ng.conf.tmpl', 'syslog-ng.conf.tmpl.jinja'],
```

Comment:
The diff of the rendering before and after these changes here:

```diff
-    - mode: 644
-    - source: salt://syslog_ng/files/syslog-ng.conf
+    - source:
+      - salt://syslog_ng/files/ABC/syslog-ng.conf.tmpl
+      - salt://syslog_ng/files/ABC/syslog-ng.conf.tmpl.jinja
+      - salt://syslog_ng/files/Debian/syslog-ng.conf.tmpl
+      - salt://syslog_ng/files/Debian/syslog-ng.conf.tmpl.jinja
+      - salt://syslog_ng/files/default/syslog-ng.conf.tmpl
+      - salt://syslog_ng/files/default/syslog-ng.conf.tmpl.jinja
```

* Was removing `mode: 644` intentional?
* Can we keep the name of the template the same as before, i.e. `syslog-ng.conf`?
* The second template option isn't useful and only adds extra I/O calls; if needed, users can set it in the pillar.

So to be clear, I think this is the ideal change for the `source`:

```diff
-    - source: salt://syslog_ng/files/syslog-ng.conf
+    - source:
+      - salt://syslog_ng/files/ABC/syslog-ng.conf
+      - salt://syslog_ng/files/Debian/syslog-ng.conf
+      - salt://syslog_ng/files/default/syslog-ng.conf
```

If that's done, then obviously the template itself will need to be renamed.